src/renderer/components/badge/badge.tsx

```diff
@@ -16,36 +16,67 @@ export interface BadgeProps extends React.HTMLAttributes<any>, TooltipDecoratorP
 @withTooltip
 @observer
 export class Badge extends React.Component<BadgeProps> {
-  @observable.ref elem: HTMLElement;
-  @observable isExpanded = false;
+  @observable _isExpanded = false;
+  @observable hasHighlightedText = false;
+  @observable.ref elem?: HTMLElement;
+
+  componentWillUnmount() {
+    document.removeEventListener("selectionchange", this.onSelectionChange)
+  }
+
+  @computed get isExpanded() {
+    return this.props.isExpanded ?? this._isExpanded
+  }
 
   @computed get isExpandable() {
-    return this.elem?.clientWidth < this.elem?.scrollWidth;
+    const { scrollWidth, clientWidth, clientHeight, scrollHeight } = this.elem || {}
+    return clientWidth < scrollWidth || clientHeight < scrollHeight
+  }
+
+  @autobind()
+  setRef(elem: HTMLElement) {
+    // This needs to be a seperate function, see: https://github.com/facebook/react/issues/11258
+    this.elem = elem
   }
 
   @autobind()
-  onMouseUp(evt: React.MouseEvent) {
-    const isTextSelected = !!document.getSelection().toString().trim();
-    if (!this.isExpandable || isTextSelected) return; // no action required
-    this.isExpanded = !this.isExpanded;
+  onSelectionChange() {
+    this.hasHighlightedText ||= document.getSelection().toString().length > 0
   }
 
   @autobind()
-  bindRef(elem: HTMLElement) {
-    this.elem = elem;
+  onMouseDown() {
+    this.onSelectionChange() // initial "event" fire on mouse down (for clearing old selections)
+    document.addEventListener("selectionchange", this.onSelectionChange)
+  }
+
+  @autobind()
+  onMouseUp() {
+    document.removeEventListener("selectionchange", this.onSelectionChange)
+
+    if (!this.hasHighlightedText) {
+      this._isExpanded = !this._isExpanded
+    }
+
+    this.hasHighlightedText = false
```

Comment:
Is it same logic what's we get in result? https://github.com/lensapp/lens/pull/1397/commits/e4e4d7efe1d9eac987897d6f7a37f8d2bd919ec8

Reply:
Not quite, see my comment on your PR